lib/Dependency.js

```diff
@@ -78,6 +79,11 @@ class Dependency {
 		this.module = null;
 	}
 }
-Dependency.compare = (a, b) => compareLocations(a.loc, b.loc);
+
+// TODO remove in webpack 5
+Dependency.compare = util.deprecate(
+	(a, b) => compareLocations(a.loc, b.loc),
+	"Dependency.compare is deprecated and will be removed in the next major version"
```

Comment:
Wouldn't it be good to explain why it will be removed?
Maybe explaining why won't be needed anymore.

Reply:
Actually this method is not used for ages. I've added a deprecation notice because some 3rd party packages may rely on it.